src/sentry/runner/commands/configoptions.py

```diff
@@ -0,0 +1,229 @@
+
+        for key in keysToFetch:
+            _get(key, dryrun)
+
+        for key, val in keysToUpdate.items():
+            _set(key, val, dryrun)
+
+        for key in keysToDelete:
+            _delete(key, dryrun)
+
+
+@configoptions.command()
+@click.argument("filename", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def strict(filename: str, dryrun: bool):
+    "Deletes everything not in the uploaded file, and applies all of the changes in the file."
+    import yaml
+
+    if dryrun:
+        click.echo("Dryrun flag on. ")
+
+    with open(filename) as stream:
+        configmap_data = yaml.safe_load(stream)
+        data = configmap_data.get("data", {}).get("options-strict.yaml", "")
+
+        for key in TRACKED:
+            if key not in data.keys():
+                _delete(key)
+
+        for key, val in data.items():
+            if key in TRACKED:
+                _set(key, val, dryrun)
+
+
+@configoptions.command()
+@click.argument("key", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def get(key: str, dryrun: bool = False) -> str:
+    "Get a configuration option."
+    return _get(key, dryrun)
+
+
+def _get(key: str, dryrun: bool = False) -> str:
+    from sentry import options
+    from sentry.options.manager import UnknownOption
+
+    try:
+        opt = options.lookup_key(key)
+        click.echo(f"Fetched Key: {opt.name} ({opt.type}) = {options.get(opt.name)}")
+        return opt
+    except UnknownOption:
+        raise click.ClickException("unknown option: %s" % key)
+    except TypeError as e:
+        raise click.ClickException(str(e))
+
+
+@configoptions.command()
+@click.argument("key", required=True)
+@click.argument("val", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def set(key: str, val: object, dryrun: bool = False) -> bool:
+    "Sets a configuration option to a new value."
+    return _set(key, val, dryrun)
+
+
+def _set(key: str, val: object, dryrun: bool = False) -> bool:
+    from sentry import options
+    from sentry.options.manager import UnknownOption
+
+    try:
+        opt = options.lookup_key(key)
+        if not dryrun:
+            options.set(key, val)
+            click.echo(f"Updated key: {opt.name} ({opt.type}) = {val}")
+            return opt
+        else:
+            click.echo(f"Updated key: {opt.name} ({opt.type}) = {val}")
+            return opt
+    except UnknownOption:
+        raise click.ClickException("unknown option: %s" % key)
+    except TypeError as e:
+        raise click.ClickException(str(e))
```

Comment:
Same concerns I raised above.

Reply:
This can happen if an option that has an int value is attempted to be set to a string value (or anything else that doesn't match type). IIRC we wanted to check for that here? But we can also check it even earlier (when generating the yaml file in the goCD repo).